File: 1725-number-of-rectangles-that-can-form-the-largest-square/1725-number-of-rectangles-that-can-form-the-largest-square.js
/**
 * @param {number[][]} rectangles
 * @return {number}
 */
var countGoodRectangles = function(rectangles) {
  const squares = rectangles.map(a => Math.min(...a))
  const max = Math.max(...squares)
  let ans = 0
  squares.forEach(l => (l === max) ? ans++ : 0)
  return ans
};

File: 1725-number-of-rectangles-that-can-form-the-largest-square/1725-number-of-rectangles-that-can-form-the-largest-square.go
func min(rect []int) int {
  if rect[0] < rect[1] {
    return rect[0]
  } else {
    return rect[1]
  }
}

func countGoodRectangles(rectangles [][]int) int {
  sq := make([]int, 0, len(rectangles))
  largest := 0
  for _, v := range rectangles {
    l := min(v)
    sq = append(sq, l)
    if l > largest {
      largest = l
    }
  }
  ans := 0
  for _, v := range sq {
    if v == largest {
      ans ++
    }
  }
  return ans
}